client/protocol.go

```diff
@@ -0,0 +1,413 @@
+package client
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	curpapi "github.com/xline-kv/go-xline/api/curp"
+	xlineapi "github.com/xline-kv/go-xline/api/xline"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+	"google.golang.org/protobuf/proto"
+)
+
+// Protocol client
+type protocolClient struct {
```

Comment:
Should this start with upper case to be used outside of this package?

Reply:
Also we should consider rename this to `Client` as there's another `ProtocolClient` in `curpapi`

Reply:
> Should this start with upper case to be used outside of this package?

I don't think so. Because protocol client is an inner client, it will only be used by other clients, should not be used by users.

Reply:
> Also we should consider rename this to `Client` as there's another `ProtocolClient` in `curpapi`

I don't think so. The code generated by protobuf doesn't contain protocol client struct. Even if, they're not in the same package and won't be a problem.

Reply:
> > Should this start with upper case to be used outside of this package?
> 
> I don't think so. Because protocol client is an inner client, it will only be used by other clients, should not be used by users.

Makes sense.


> > Also we should consider rename this to `Client` as there's another `ProtocolClient` in `curpapi`
> 
> I don't think so. They're not in the same package and won't be a problem.

Duplicate names can lead to confusion, even they are in different modules, I think it's better renaming it to `curpClient` or just `client`.

Reply:
done